frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -799,10 +831,25 @@ pub mod pallet {
                 ));
             }
 
-            T::Currency::resolve_creating(&staker, reward_imbalance);
+            // Reward willbe transferred to either Staker or the account staker delegated to
+            let mut final_reward_claimee = staker.clone();
+
+            if reward_destination == RewardDestination::DelegateBalance {
+                final_reward_claimee = Self::check_delegate(staker.clone(), contract_id.clone())?;
+            }
```

Comment:
Just a suggestion, but a more _Rust_ like way would be:
```Rust
let reward_beneficiary = if reward_destination = RewardDestination::DelegateBalance {
    staker.clone()
} else {
    Self::check_delegate(staker.clone(), contract_id.clone())?
};
```

Reply:
Looks more idiomatic, will keep this in mind.